instructions/humus-rest.instructions.md

```diff
@@ -0,0 +1,451 @@
+---
+description: 'Patterns and best practices for REST API applications using Humus'
+applyTo: '**/*.go'
+---
+
+# Humus Framework - REST Service Instructions
+
+This file provides patterns and best practices specific to REST API applications using Humus. Use this file alongside `humus-common.instructions.md` for complete guidance.
+
+## Project Structure
+
+Use this structure for production services. This matches the examples in the Humus repository:
+
+```
+my-rest-service/
+├── main.go              # Minimal entry point (just calls app.Init)
+├── config.yaml          # Configuration
+├── app/
+│   └── app.go          # Init function and Config type
+├── endpoint/           # REST endpoint handlers
+│   ├── create_user.go
+│   ├── get_user.go
+│   └── list_users.go
+├── service/            # Backend service clients
+│   ├── user.go         # User service client
+│   └── order.go        # Order service client
+├── go.mod
+└── go.sum
+```
+
+**main.go:**
+```go
+package main
+
+import (
+    "bytes"
+    _ "embed"
+    "github.com/z5labs/humus/rest"
+    "my-service/app"
+)
+
+//go:embed config.yaml
+var configBytes []byte
+
+func main() {
+    rest.Run(bytes.NewReader(configBytes), app.Init)
+}
+```
+
+**app/app.go:**
+```go
+package app
+
+import (
+    "context"
+    "my-service/endpoint"
+    "github.com/z5labs/humus/rest"
+)
+
+type Config struct {
+    rest.Config `config:",squash"`
+    // Add service-specific config here
+}
+
+func Init(ctx context.Context, cfg Config) (*rest.Api, error) {
+    api := rest.NewApi(
+        cfg.OpenApi.Title,
+        cfg.OpenApi.Version,
+        endpoint.CreateUser(),
+        endpoint.GetUser(),
+        endpoint.ListUsers(),
+    )
+    return api, nil
+}
+```
+
+## Configuration
+
+**Custom Config with provider interface:**
+
+If you need to customize the HTTP server listener (e.g., custom port), implement the `ListenerProvider` interface:
+
+```go
+type Config struct {
+    rest.Config `config:",squash"`
+    
+    HTTP struct {
+        Port uint `config:"port"`
+    } `config:"http"`
+}
+
+func (c Config) Listener(ctx context.Context) (net.Listener, error) {
+    return net.Listen("tcp", fmt.Sprintf(":%d", c.HTTP.Port))
+}
+```
+
+See `humus-common.instructions.md` for general configuration patterns like using Go templates in YAML.
+
+## Backend Service Clients
```

Comment:
After thinking about it this should be in the common instructions because this applies to any humus application type. It is a common pattern for organizing backend service clients usable in any application